packages/react-scripts/fixtures/yarn-ws/ws/package.json

```diff
@@ -0,0 +1,4 @@
+{
+  "private": true,
+  "workspaces": ["packages/*"]
```

Comment:
What happens if somebody specifies something like `"workspaces": ["src/*"]`? Will this work? Are there any gotchas?

If something like this could break our setup I’m in favor of being more restrictive (e.g. I don’t mind completely forbidding specifying anything other than `"packages/*"`).

What are your thoughts? Can you see any values for `workspaces` that would break our assumptions?

Reply:
Not related to workspaces setting, but this made me think...if apps / shared components specify different versions of react?  I think multiple versions of react could get packaged and not only cause bloat, but things will not work?  Maybe there are only edge cases where this could be an issue...not sure, probably need to go through some scenarios, could be pretty painful for a user to debug unless there's something that already detects that somehow?

Reply:
We definitely need to break the build on multiple versions of React (since it doesn't work in very subtle ways). But it's not unique to this PR. See https://github.com/facebookincubator/create-react-app/issues/675. We'll want to merge some solution before 2.0 anyway so this isn't blocked on it.